docs/adr/0013-react-toolchain-for-settings-redesign.md

```diff
@@ -0,0 +1,118 @@
+
+Firefox Accounts is introducing a new visual design for Settings (the interface that exists at the address `https://accounts.firefox.com/settings`). The changes involved are significant enough such that it [has been decided](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md) that an entirely new React App will be built more or less independent of the existing Settings web app. With this we can beta test and roll out the new design with minimal disruption.
+
+As there are a multitude of ways a React App can be set up and configured, this decision has introduced the question, "What [toolchain](https://en.wikipedia.org/wiki/Toolchain) should we set up a new React App with?". This ADR serves to answer that question by going over the various approaches we can take to set up a new React App and how it might integrate with our existing FxA ecosystem.
+
+It's important to note that we are not deciding the languages, libraries, or other tools that we will use within the React development environment, but rather, the system that will be used to develop the React App with.
+
+Finally, we are not making decisions about the App's production server here. We'll be using [fxa-content-server](https://github.com/mozilla/fxa/tree/master/packages/fxa-content-server/server)'s Express server for the production build.
+
+## Decision Drivers
+
+- **Familiarity** - unless there is an overwhelming reason, we don't really want to introduce a completely foreign build process for everyone to learn.
+- **Configurability** - if it is less configurable does it meet our needs; if it is more configurable is the time spent worth it?
+- **Maintainability** - how much work do we set up for ourselves if we're maintaining all upgrades and configuration?
+
+Additionally, our team has a desire to share React components and other files across projects. While this ADR does not specifically address this, it will be taken into consideration.
+
+## Considered Options
+
+- Option A - Completely Custom Approach
+- Option B - Create React App
+  - Option B2 - Create React App and immediately eject
+  - Option B3 - Create React App and use Rescripts
+- Option C - Neutrino with React Preset
+- Option D - Other Considerations
+
+## Decision Outcome
+
+Chosen option: "B3 - Create React App and use Rescripts", because:
+
+- Our team has sufficient knowledge in working with CRA over the other options.
+- We don't need to re-invent the wheel. Our needs for the Settings project are not so great that it would be worth spending time creating a ground-up toolchain. To some extent this option allows us to do what we do best without needing to worry about the specifics of the development workflow.
+- While not completely configurable compared to other options, the addition of Rescripts can provide us with greater flexibility where we might need it (and can possibly help us with the external imports restriction CRA has).
```

Comment:
I think it would be ideal to determine if rescripts will actually allow us to import outside of `src/`. We have an issue for "share components" in #4817, but that really plays into the decision here. A quick search lead me to https://github.com/oklas/react-app-rewire-alias, I haven't poked at that much but... Maybe the solution is (especially if we're going to use CSS vars over SCSS vars) CRA + that `rewire-alias` tool, which seems to allow for multiple `src` dirs? Or maybe rescripts allows this functionality and has better support than `rewire-alias`?

Reply:
rescripts has a function specifically for removing webpack plugins: https://github.com/harrysolovay/rescripts#removewebpackpluginconstructorname-config  Looks simple enough.

Reply:
That LGTM. @jodyheavener I also came across https://github.com/gsoft-inc/craco, looks similar to rescripts with possibly better support?

Reply:
@LZoog @chenba I've built out a solution that addresses and demonstrates this; will update the ADR to reflect.